pkgs/misc/vim-plugins/overrides.nix

```diff
@@ -382,6 +383,30 @@ self: super: {
     dependencies = with self; [ plenary-nvim ];
   });
 
+  markdown-preview-nvim = super.markdown-preview-nvim.overrideAttrs (old: let
+    # We only need its dependencies `node-modules`.
+    nodeDep = nodePackages."markdown-preview-nvim-../../misc/vim-plugins/markdown-preview-nvim".overrideAttrs (old: {
+      dontNpmInstall = true;
+    });
+  in {
+    patches = [
+      (substituteAll {
+        src = ./markdown-preview-nvim/fix-node-paths.patch;
+        node = "${nodejs}/bin/node";
+      })
+    ];
+    postInstall = ''
+      # The node package name is `*-vim` not `*-nvim`.
+      ln -s ${nodeDep}/lib/node_modules/markdown-preview-vim/node_modules $out/app
```

Comment:
Lets move that to some FHS directory and not create /app.

Reply:
You should build it first and see what's the current behavior.
All vimPlugins follows vim's package hierarchy, with special meaning directories like `autoload`, `plugin` and etc directly under `$out`. The FHS prefix is removed in https://github.com/NixOS/nixpkgs/commit/56f823dd5c596ef6374f99a22ca63168ff6f6fb9

`app` is a directory from upstream containing a node package. This `ln` creates `$out/app/node_modules`